lib/rspec/core/configuration.rb

```diff
@@ -870,7 +874,7 @@ def full_description
       # Adds a formatter to the set RSpec will use for this run.
       #
       # @see RSpec::Core::Formatters::Protocol
-      def add_formatter(formatter, output=output_stream)
+      def add_formatter(formatter, output=output_wrapper)
```

Comment:
I think we should take the given output stream and wrap it, the way this is currently it will allow someone to not use a wrapper and thus re trigger the bug